catkin_tools/context.py

```diff
@@ -49,19 +49,12 @@ class Context(object):
     This context can be locked, so that changing the members is prevented.
     """
 
-    DEFAULT_LOG_SPACE = 'logs'
-    DEFAULT_SOURCE_SPACE = 'src'
-    DEFAULT_BUILD_SPACE = 'build'
-    DEFAULT_DEVEL_SPACE = 'devel'
-    DEFAULT_INSTALL_SPACE = 'install'
+    CATKIN_SPACES_GROUP = 'catkin_tools.spaces'
+
+    SPACES = {}
```

Comment:
Given the usage patterns seen below, there might be a case to be made here for using a list rather than a dict, and creating a new `Space` class rather than just assigning the entry_point dict directly.

I don't think it much matters for now, so my inclination would be to leave it and add the extra layer as need demands.